packages/govuk-frontend/src/govuk/components/components.template.test.js

```diff
@@ -65,10 +66,14 @@ describe('Components', () => {
       // Validate component examples
       for (const { component: componentName, fixtures } of componentsFixtures) {
         const fixtureTasks = fixtures.map(async (fixture) => {
-          const html = render(componentName, {
-            context: fixture.options,
-            fixture
-          })
+          const html = outdent`
+            <div id="content" class="govuk-width-container">
+              ${render(componentName, {
+                context: fixture.options,
+                fixture
+              })}
+            </div>
+          `
```

Comment:
I think this could be slightly misleading because this looks very much like one of our main wrapper elements, which I don't think is something you'd ever associate with a component using `aria-describedby`.

Can we make it clearer what this is for by:

- adding the extra element _after_ the rendered component
- changing the ID to something like `test-target-element` (and in the corresponding examples)
- adding a comment to explain what it's for

```suggestion

          const html = outdent`
            ${render(componentName, {
              context: fixture.options,
              fixture
            })}
            <div id="test-target-element">
              <!-- Target for references in examples (e.g. aria-describedby) so no-missing-references can pass -->
            </div>
          `
```

Reply:
@36degrees Mind if I tag @romaricpascal in?

I thought the same as you but we agreed it was acceptable for the Skip link accessibility tests

I'd like to make your suggestion there too

Reply:
Not sure I follow, sorry – how is the skip link involved here?

Reply:
Ah I'll explain, it was covered in dev catch-up after your holiday

E.g. **Skip link** failed Accessibility tests (JavaScript error) without a target element

So we decided that a dummy element ID was acceptable for now, but due to hard-coded defaults like `default("#content", true)` we [added `id="content"` to the component preview wrapper](https://github.com/alphagov/govuk-frontend/commit/3ed9d1f418ded73ae4e76cc88835b5f1f258db72) just like in this PR

I like your suggestion so perhaps we can annotate the test ID in both wrappers?

Reply:
Here's the existing component preview wrapper:

https://github.com/alphagov/govuk-frontend/blob/3ed9d1f418ded73ae4e76cc88835b5f1f258db72/shared/lib/components.js#L209-L211

Reply:
Thanks, that's helpful context 👍🏻 

I think annotating the test ID in both wrappers sounds like a good solution.

Reply:
Morning 👋

I've swapped all the `#content` references in 3ed9d1f418ded73ae4e76cc88835b5f1f258db72 to `test-target-element`

Whilst the HTML validation tests lack the `<div id="content">` wrapper, it's still needed by:

```console
npx jest --selectProjects "Accessibility tests"
npx jest --selectProjects "JavaScript component tests"
```

But that's only because it's hard coded in [`href="{{ params.href | default("#content", true) }}"`](https://github.com/alphagov/govuk-frontend/blob/545fde1dda58d2b502d3a67618ebd0dc9567ee51/packages/govuk-frontend/src/govuk/components/skip-link/template.njk#L1)

Without it we'll see JavaScript warnings for every Skip link example:

```console
  console.warn
    Initialising `new SkipLink()` with example 'no href' threw:
     ElementError: Skip link: Target content (`id="content"`) not found
```

Think we're looking good now, ready for review again 👍